client/src/components/pages/demo/track/Waveform.js

```diff
@@ -0,0 +1,122 @@
+// This waveform was made with the help of this guide:
+// https://css-tricks.com/making-an-audio-waveform-visualizer-with-vanilla-javascript/
+import React, { useEffect, useRef, useState } from 'react'
+import { Scrubber } from 'components/pages/demo/Scrubber';
+
+export const Waveform = ({
+  track,
+  demoLength,
+  timeState: [currentTime, setCurrentTime],
+  playingState: [playing, setPlaying]
+}) => {
+  let trackStart;
+  let trackWidth;
+

```

Comment:
This is how you could dynamically get the updated value of the scrubber height so you could use that for the canvas.

```suggestion
  const [ trackHeight, setTrackHeight ] = useState(0);
  const scrubberRef = useRef(null);


  const setHeightOnResize = () => {
    setTrackHeight(scrubberRef.current.offsetHeight);
  }

  useEffect(() => {

    window.addEventListener('resize', setHeightOnResize);

    setTrackHeight(scrubberRef.current.offsetHeight);

    return () => window.removeEventListener('resize', setHeightOnResize);

  }, []);
```


```suggestion
    // set scrubber ref
  <Scrubber 
        timeState={[currentTime, setCurrentTime]}
        demoLength={demoLength}
        playingState={[playing, setPlaying]}
        scrubberType={"trackScrubber"}
        ref={scrubberRef}
      />
      
      ...
      // pass in trackHeight here to use for canvas
      <RenderedWaveform filteredData={filteredData} trackWidth={trackWidth} trackHeight={trackHeight} />
```